cmd/clusterctl/client/cluster/internal/dryrun/client.go

```diff
@@ -0,0 +1,330 @@
+	panic("implement me")
+}
+
+// Status returns a client which can update status subresource for kubernetes objects.
+func (c *Client) Status() client.StatusWriter {
+	return c.fakeClient.Status()
+}
+
+// Scheme returns the scheme this client is using.
+func (c *Client) Scheme() *runtime.Scheme {
+	return c.fakeClient.Scheme()
+}
+
+// RESTMapper returns the rest this client is using.
+func (c *Client) RESTMapper() meta.RESTMapper {
+	return c.fakeClient.RESTMapper()
+}
+
+// Changes generates a summary of all the changes observed since the creation of the client
+// to when this function is called.
+func (c *Client) Changes(ctx context.Context) (*Changes, error) {
+	changes := &Changes{
+		Created:  []*unstructured.Unstructured{},
+		Modified: []*Modification{},
+		Deleted:  []*unstructured.Unstructured{},
+	}
+
+	// Handle the created objects.
+	for id := range c.changeTracker.created {
+		obj := &unstructured.Unstructured{}
+		obj.SetGroupVersionKind(id.gvk)
+		obj.SetNamespace(id.key.Namespace)
+		obj.SetName(id.key.Name)
+		if err := c.fakeClient.Get(ctx, id.key, obj); err != nil {
+			return nil, errors.Wrapf(err, "failed to read created object %s", id.key.String())
+		}
+		changes.Created = append(changes.Created, obj)
+	}
+
+	// Handle the modified objects.
+	for id, original := range c.changeTracker.modified {
+		after := &unstructured.Unstructured{}
+		after.SetGroupVersionKind(id.gvk)
+		after.SetNamespace(id.key.Namespace)
+		after.SetName(id.key.Name)
+		if err := c.fakeClient.Get(ctx, id.key, after); err != nil {
+			return nil, errors.Wrapf(err, "failed to read modified object %s", id.key.String())
+		}
+		patch := client.MergeFrom(original)
+		diff, err := patch.Data(after)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to calculate diff of modified object %s", id.key.String())
+		}
+		before := &unstructured.Unstructured{}
+		if err := c.Scheme().Convert(original, before, nil); err != nil {
+			return nil, errors.Wrapf(err, "failed to convert %s to unstructured", client.ObjectKeyFromObject(original).String())
+		}
+		changes.Modified = append(changes.Modified, &Modification{
+			Before: before,
+			After:  after,
+			Diff:   diff,
+		})
+	}
+
+	// Handle the deleted objects.
+	for id := range c.changeTracker.deleted {
+		obj := &unstructured.Unstructured{}
+		obj.SetGroupVersionKind(id.gvk)
+		obj.SetNamespace(id.key.Namespace)
+		obj.SetName(id.key.Name)
+		changes.Deleted = append(changes.Deleted, obj)
+	}
+
+	return changes, nil
+}
+
+// ensureObjInFakeClient makes sure that the object is available in the fake client.
+// If the object is not already availbale it will add it to the fake client by running a "Create"
+// operation.
+func (c *Client) ensureObjInFakeClient(ctx context.Context, obj client.Object) error {
+	o := obj.DeepCopyObject().(client.Object)
+	if err := c.fakeClient.Get(ctx, client.ObjectKeyFromObject(o), o); err != nil {
```

Comment:
I think directly calling create and ignoring the AlreadyExists error should be simpler and easier to read